traitsui/ui.py

```diff
@@ -93,7 +94,12 @@ class UI(HasPrivateTraits):
     history = Any()
 
     #: The KeyBindings object (if any) for this UI:
-    key_bindings = Property(depends_on=["view._key_bindings", "context"])
+    key_bindings = Property(
+        observe=[
+            trait("view._key_bindings", optional=True),
+            "context",
+        ]
+    )
```

Comment:
I think this use of `trait` may be broken (but break is hidden by `optional=True`)
see https://github.com/enthought/pyface/pull/892 and https://github.com/enthought/pyface/pull/880#discussion_r573969099

Reply:
I removed this to go ahead and merge this PR. We can address the few % of on trait change uses later.